cohortextractor/patients.py

```diff
@@ -397,11 +398,197 @@ def mean_recorded_value(
                     "incidence": 0.95,
                 },
             )
+
+        This creates a variable returning a float of the mean recorded creatine level
+        creatinine level over a 6 month period:
+
+        creatinine=patients.mean_recorded_value(
+            creatinine_codes,
+            on_most_recent_day_of_measurement=False,
+            between=["2019-09-16", "2020-03-15"],
+            return_expectations={
+                "float": {"distribution": "normal", "mean": 150, "stddev": 200},
+                "date": {"earliest": "2019-09-16", "latest": "2020-03-15"},
+                "incidence": 0.75,
+            },
+        ),
```

Comment:
If a researcher were to add `include_measurement_date=True` and `date_format="YYYY-MM-DD`, what date would be returned? (Similarly for month and year, especially when the range spans months and years.)

Reply:
In this example, because `on_most_recent_day_of_measurement=False`, it would raise an exception (an AssertionError), because it doesn't make any sense to inclde the measurement date if the mean is taken over more than one day.  I didn't add that to the docstring because there are various other places where cohortextractor just asserts that invalid kwargs aren't included, without explicitly documenting it.  It'll fail immediately on a local run, with a message, so it should be obvious what's wrong.
 
If `on_most_recent_day_of_measurement=True` and `include_measurement_date=True`, it'll return the single date that was the most recent day of measurement.  If the date format was `YYYY`, it'll just return that day with year, which won't necessarily be any use to anyone if the range is a number of months within the same year, but I guess that's up to the individual researcher.

There's also another way to get the date.  If you leave out the i`nclude_measurement_date=True` (or make it False), you can define another variable to get the date, by doing `patients.date_of(creatinine, date_format="YYYY=MM-DD")`  (as we do in the tests), which also lets you give it a custom name.  I'm guessing that this is the more recent way to do it, but I followed the documentation of the existing `mean_recorded_value`.  Maybe it makes sense to include that option in the docstrings too.